R/checkModels.R

```diff
@@ -1,74 +1,203 @@
 ### called by splitModels() and/or defineModel().
 ### Contains all functions that check certain aspects of models.
 
+### called by splitModels() ----------------------------------------------------
+
+checkPersonGroupsConsistency <- function(d){
+  d <- eatTools::makeDataFrame(d, verbose=FALSE)
+  ### Eintraege in erster Spalte muessen unique sein und duerfen keine missings enthalten
+  if(any(is.na(d[,1]))){
+    stop("Person identifier in first column of 'person.groups' has missing values.")}
+  if(length(d[,1]) != length(unique(d[,1]))){
+    stop("Person identifier in first column of 'person.groups' is not unique.")}
+  ### die naechsten checks erfolgen jeweils fuer alle weiteren spalten
+  chk1 <- lapply(colnames(d)[-1], FUN = function (x){
+    ### gruppierungsvariablen duerfen nicht konstant sein und keine fehlenden Werte haben
+    if(length(unique(d[,x])) == 1){
+      stop(paste0("Column '",x,"' of 'person.groups' is constant."))}
+    if(any(is.na(d[,x]))){
+      stop(paste0("Column '",x,"' of 'person.groups' has missing values."))}
+  })
+
+  # wenn nur eine Spalte wird diese als IDs angenommen
+  if(checkmate::test_data_frame(person.groups, ncols = 1)){
```

Comment:
This will fail. The argument `person.groups` is not defined within `checkPersonConsistency()`. You would have to replace `person.groups` with `d`. However, I would recommend to replace lines 22 - 35 with something like `checkmate::test_data_frame(d, min.cols = 2)`
The test `# hat erste Spalte mehr Elemente als alle anderen` is already captured in line 9-10
```r
  if(length(d[,1]) != length(unique(d[,1]))){
    stop("Person identifier in first column of 'person.groups' is not unique.")}
```
This can also be removed, too.